homeassistant/components/geniushub/climate.py

```diff
@@ -123,34 +159,52 @@ def operation_list(self):
     @property
     def current_operation(self):
         """Return the current operation mode."""
-        return GH_STATE_TO_HA.get(self._objref.mode)
+        return GH_STATE_TO_HA[self._zone.mode]
 
     @property
     def is_on(self):
         """Return True if the device is on."""
-        return self._objref.mode in GH_STATE_TO_HA
+        return self._zone.mode in GH_STATE_TO_HA
 
     async def async_set_operation_mode(self, operation_mode):
         """Set a new operation mode for this zone."""
-        await self._objref.set_mode(HA_OPMODE_TO_GH.get(operation_mode))
+        await self._zone.set_mode(HA_OPMODE_TO_GH[operation_mode])
 
     async def async_set_temperature(self, **kwargs):
         """Set a new target temperature for this zone."""
         temperature = kwargs.get(ATTR_TEMPERATURE)
-        await self._objref.set_override(temperature, 3600)  # 1 hour
+        await self._zone.set_override(temperature, 3600)  # 1 hour
 
     async def async_turn_on(self):
-        """Turn on this heating zone."""
-        await self._objref.set_mode(HA_OPMODE_TO_GH.get(STATE_AUTO))
+        """Turn on this heating zone.
+
+        Set Zones to Footprint mode if they have a Room sensor, and to Timer
+        mode otherwise.
+        """
+        mode = STATE_ECO if hasattr(self._zone, 'occupied') else STATE_AUTO
+        await self._zone.set_mode(HA_OPMODE_TO_GH[mode])
 
     async def async_turn_off(self):
         """Turn off this heating zone (i.e. to frost protect)."""
-        await self._objref.set_mode(GH_OPMODE_OFF)
+        await self._zone.set_mode(HA_OPMODE_TO_GH[STATE_OFF])
+
+
+class GeniusClimateHub(GeniusClimateBase):
```

Comment:
This shouldn't be an entity. It should be a separate class, in the geniushub component, that handles the updates and the data from the api and notifies the entities when there is new data.

See eg the mopar component.
https://github.com/home-assistant/home-assistant/blob/118d3bc11cf8a94efbc7ffd9afb73d225ed9d5c1/homeassistant/components/mopar/__init__.py#L106-L126

Reply:
Done.